File: ps/rewyear/leetcode/7.Reverse_Integer.java
class Solution {
    public int reverse(int x) {
        StringBuffer str = new StringBuffer(Integer.toString(x));
        
        str.reverse();
        
        if(str.charAt(str.length()-1) == '-'){
            str.deleteCharAt(str.length()-1); // delete minus 
            str.insert(0,'-'); // add minus
        }
            
        long reversed = Long.parseLong(str.toString());
        if(reversed < (int)-2e31 || reversed > (int)2e31 - 1)
            return 0;
        else
            return (int)reversed;
    }
}

File: zziri/ps/leetcode/Problems/7. Reverse Integer.py
class Solution {
    public int reverse(int x) {
        int minus = 1;
        if (x < 0) {
            minus = -1;
            x = -x;
        }
        long ret = 0;
        while (x > 0) {
            ret *= 10;
            ret += x % 10;
            x /= 10;
        }
        ret *= minus;
        if (ret < (int)-2e31 || ret > (int)2e31-1)
            return 0;
        else
            return (int)ret;
    }
}
